# Work around an upstream Undici issue when running in the Electron renderer
# process.
# See https://github.com/nodejs/undici/issues/3997
diff --git a/lib/dispatcher/client-h1.js b/lib/dispatcher/client-h1.js
index 9f6f17c1579be03c90a0800653e731633d2bccb3..8c6b5e3b1b8824c8c7f13692e3b023ae4232ab54 100644
--- a/lib/dispatcher/client-h1.js
+++ b/lib/dispatcher/client-h1.js
@@ -6,6 +6,8 @@ const assert = require('node:assert')
 const util = require('../core/util.js')
 const { channels } = require('../core/diagnostics.js')
 const timers = require('../util/timers.js')
+// PATCH: Electron's setTimeout does not have an ".unref" method.
+const { setTimeout, clearTimeout } = require('node:timers');
 const {
   RequestContentLengthMismatchError,
   ResponseContentLengthMismatchError,
diff --git a/lib/util/timers.js b/lib/util/timers.js
index 14984d42ef2901af503b595002bc973bf069f9a4..3c4ea30330dbcace51069f84afbd03cfa2c20710 100644
--- a/lib/util/timers.js
+++ b/lib/util/timers.js
@@ -1,5 +1,9 @@
 'use strict'
 
+// PATCH: Electron's setTimeout does not have an ".unref" method.
+// See https://github.com/nodejs/undici/issues/3997
+const { setTimeout, clearTimeout } = require('node:timers');
+
 /**
  * This module offers an optimized timer implementation designed for scenarios
  * where high precision is not critical.
diff --git a/lib/web/eventsource/eventsource.js b/lib/web/eventsource/eventsource.js
index 17a1de7b7ba6bcc6f27a5ad60fecfe458316279c..23e2af31127e13324b2d22af47b747fcce366114 100644
--- a/lib/web/eventsource/eventsource.js
+++ b/lib/web/eventsource/eventsource.js
@@ -10,6 +10,9 @@ const { isNetworkError } = require('../fetch/response')
 const { kEnumerableProperty } = require('../../core/util')
 const { environmentSettingsObject } = require('../fetch/util')
 const { createPotentialCORSRequest } = require('./util')
+// PATCH: Electron's setTimeout does not have an ".unref" method.
+const { setTimeout, clearTimeout } = require('node:timers');
+
 
 let experimentalWarned = false
 
@@ -346,7 +349,7 @@ class EventSource extends EventTarget {
 
       //   4. Fetch request and process the response obtained in this fashion, if any, as described earlier in this section.
       this.#connect()
-    }, this.#state.reconnectionTime)?.unref()
+    }, this.#state.reconnectionTime)?.unref?.()
   }
 
   /**
diff --git a/lib/web/fetch/index.js b/lib/web/fetch/index.js
index 935bc9d4c90e167fc7b8edaf944edeaf6f28d2fc..f157a413eaff3fbe4d0c11745913d8a33d676cc9 100644
--- a/lib/web/fetch/index.js
+++ b/lib/web/fetch/index.js
@@ -13,6 +13,7 @@ const {
 const { HeadersList } = require('./headers')
 const { Request, cloneRequest, getRequestDispatcher, getRequestState, removeRequestAbortListener } = require('./request')
 const zlib = require('node:zlib')
+const { performance } = require('node:perf_hooks');
 const {
   makePolicyContainer,
   clonePolicyContainer,
